AmplifyPlugins/API/Tests/AWSAPIPluginTests/Support/Decode/GraphQLResponseDecoder+DecodeDataTests.swift

```diff
@@ -110,12 +110,8 @@ extension GraphQLResponseDecoderTests {
 
         let result = try decoder.decodeToResponseType(graphQLData)
         XCTAssertNotNil(result)
-        let fetchCompleted = expectation(description: "Fetch completed")
-        result.fetch { _ in
-            XCTAssertEqual(result.count, 2)
-            fetchCompleted.fulfill()
-        }
-        wait(for: [fetchCompleted], timeout: 1)
+        try await result.fetch()
```

Comment:
Thanks for pointing this out Brennan, I'm updating this test to match the pattern you have provided there, and going through the changes in this PR to see where else I can add this code. 

I'm assuming when you say unit tests, you mean integration tests as well. I believe integration tests like in the APIPlugin will benefit greatly from this pattern since it's backed by a network call. We can then timeout on receiving network response, instead of hanging the integration tests. 

I'm not 100% convinced this is useful for unit tests since if new code changes introduce an hanging test, we will catch that during the local unit test run, and then the PR pipeline workflows, whereas hanging integration tests will cause the workflow to run forever.

I'm not sure what is the state of other plugins test code, and think DataStore is the first plugin to remove the listener APIs. So I opened a follow up task to make sure we do this for all plugins later https://github.com/aws-amplify/amplify-ios/issues/2216

Reply:
Just updated the PR after going through and applying the suggested wrapping of async calls. I updated most of the unit tests that were changed in this PR to follow the pattern in this [commit](https://github.com/aws-amplify/amplify-ios/pull/2214/commits/7e1c268b50d097bcfe920b14929ab9973f74606b). I left the integration tests as is, so there's a potential that the integration tests may hang and not timeout on async calls. DataStore performs SQL operations locally so I'm less worried that this is the case, but feel making the lift now will be out of scope of this PR. Since the integration tests does not impact the main PR pipeline, are passing as is (not hanging), i'm going to track in https://github.com/aws-amplify/amplify-ios/issues/2216 and defer this work right now.